src/algebra/ordered_group.lean

```diff
@@ -93,482 +101,683 @@ instance ordered_comm_group.has_exists_mul_of_le (α : Type u)
+lemma le_mul_of_mul_inv_le_right (h : a * c⁻¹ ≤ b) : a ≤ b * c :=
+mul_inv_le_iff''.mp h
+/- ...to here. -/
+
+end covariant_le_right
+
+section covariant_lt_right
+variables [covariant_class α α (function.swap (*)) (<)] {a b c : α}
+
+@[simp, to_additive]
+lemma mul_lt_mul_iff_right' (c : α) : a * c < b * c ↔ a < b :=
 begin
-  have h := mul_le_mul_left' h b,
-  rwa mul_inv_cancel_left at h
+  refine ⟨λ h, _, λ h, mul_lt_mul_right_n _ h⟩,
+  obtain F : a * c * c⁻¹ < b * c * c⁻¹ := mul_lt_mul_right_n c⁻¹ h,
+  simpa only [mul_inv_cancel_right] using F,
 end
 
+
+@[simp, to_additive]
+lemma div_lt_div_iff_right (c : α) : a / c < b / c ↔ a < b :=
+by { rw [← mul_lt_mul_iff_right' c], simp [div_eq_mul_inv] }
+
+/- The following lemma is probably superfluous. -/
 @[to_additive]
-lemma inv_mul_le_of_le_mul (h : a ≤ b * c) : b⁻¹ * a ≤ c :=
-begin
-  have h := mul_le_mul_left' h b⁻¹,
-  rwa inv_mul_cancel_left at h
-end
+lemma div_lt_div_right'' (h : a < b) (c : α) : a / c < b / c :=
+(div_lt_div_iff_right c).2 h
+
+@[simp, to_additive sub_pos]
+lemma one_lt_div'' : 1 < a / b ↔ b < a :=
+by rw [← mul_lt_mul_iff_right' b, one_mul, div_eq_mul_inv, inv_mul_cancel_right]
+
+alias sub_pos ↔ lt_of_sub_pos sub_pos_of_lt
+
+@[simp, to_additive sub_lt_zero]
+lemma div_lt_one'' : a / b < 1 ↔ a < b :=
+by rw [← mul_lt_mul_iff_right' b, one_mul, div_eq_mul_inv, inv_mul_cancel_right]
 
+alias sub_lt_zero ↔ lt_of_sub_neg sub_neg_of_lt
+
+/- This lemma has a slightly awkward `mul_inv` name and is intended as a stepping stone towards the
+next lemma `lt_div_iff_mul_lt`. -/
 @[to_additive]
-lemma le_mul_of_inv_mul_le_left (h : b⁻¹ * a ≤ c) : a ≤ b * c :=
-le_mul_of_inv_mul_le h
+lemma lt_mul_inv_iff_mul_lt : b < c * a⁻¹ ↔ b * a < c :=
+by rw [← mul_lt_mul_iff_right' a, inv_mul_cancel_right]
 
+/- The primed version of this lemma involves commuting `b` and `a`: `b < c / a ↔ a * b < c`.
+It is proven later, when there is a commutativity assumption on the group. -/
 @[to_additive]
-lemma inv_mul_le_left_of_le_mul (h : a ≤ b * c) : b⁻¹ * a ≤ c :=
-inv_mul_le_of_le_mul h
+lemma lt_div_iff_mul_lt : b < c / a ↔ b * a < c :=
+by rw [div_eq_mul_inv, lt_mul_inv_iff_mul_lt]
 
+/- This lemma has a slightly awkward `mul_inv` name and is intended as a stepping stone towards the
+next lemma `div_lt_iff_lt_mul`.  The primed name is for consistency with earlier use. -/
 @[to_additive]
-lemma le_mul_of_inv_mul_le_right (h : c⁻¹ * a ≤ b) : a ≤ b * c :=
-by { rw mul_comm, exact le_mul_of_inv_mul_le h }
+lemma mul_inv_lt_iff_lt_mul' : a * b⁻¹ < c ↔ a < c * b :=
+by rw [← mul_lt_mul_iff_right' b, inv_mul_cancel_right]
 
+/- The non-primed version of this lemma involves commuting `c` and `b`: `a / b < c ↔ a < b * c`.
+It is proven later, when there is a commutativity assumption on the group. -/
 @[to_additive]
-lemma inv_mul_le_right_of_le_mul (h : a ≤ b * c) : c⁻¹ * a ≤ b :=
-by { rw mul_comm at h, apply inv_mul_le_left_of_le_mul h }
+lemma div_lt_iff_lt_mul' : a / b < c ↔ a < c * b :=
```

Comment:
```suggestion
lemma div_lt_iff_lt_mul : a / b < c ↔ a < c * b :=
```
makes this match the name the lemma currently has in master.

Reply:
Ok, I think that I am slowly understanding what I did wrong: I replaced automatically the auto-generated name, but then did not change the actual name of the lemma!

Reply:
Alright, I hope that I have fixed this issue now!  I also learned about the `git push ... ... --force` command, that I hope I used correctly!